src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,109 @@
+import mate.academy.springboot.swagger.services.ProductService;
+import mate.academy.springboot.swagger.services.mapper.ProductMapper;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+@RequiredArgsConstructor
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+
+    @PostMapping
+    @ApiOperation(value = "create product")
+    public ProductResponseDto create(@RequestBody
+                                     ProductRequestDto productRequestDto) {
+        return productMapper.toDto(
+                productService.crete(
+                        productMapper.toModel(productRequestDto)));
+    }
+
+    @GetMapping("/{id}")
+    @ApiOperation(value = "get product by id")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.toDto(productService.getById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    @ResponseStatus(HttpStatus.OK)
+    @ApiOperation(value = " delete product by id")
+    public void deleteById(@PathVariable Long id) {
+        productService.deleteByIud(id);
+    }
+
+    @PutMapping("/{id}")
+    @ApiOperation(value = "update product by id")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @RequestBody ProductRequestDto productRequestDto) {
+        Product product = productMapper.toModel(productRequestDto);
+        product.setId(id);
+        return productMapper.toDto(productService.update(product));
+    }
+
+    @GetMapping
+    @ApiOperation(value = "get all product")
+    public List<ProductResponseDto> getAll(@ApiParam(value = "default value is 10")
+                                           @RequestParam(defaultValue = "10") Integer count,
+                                           @ApiParam(value = "default value is 0")
+                                           @RequestParam(defaultValue = "0") Integer page,
+                                           @RequestParam String sortBy) {
+        Sort sort;
+        if (sortBy.equals("title")) {
+            sort = Sort.by(Sort.Direction.DESC, sortBy);
+        } else {
+            sort = Sort.by(sortBy);
+        }
```

Comment:
you have the same logic below, you should avoid duplication + both title and price can be sorted in DESC or ASC